workflow/controller/operator.go

```diff
@@ -2705,6 +2711,15 @@ func (woc *wfOperationCtx) loadExecutionSpec() (wfv1.TemplateReferenceHolder, wf
 		}
 		woc.wf.Status.StoredWorkflowSpec = wftSpec
 		woc.updated = true
+	} else if woc.controller.referenceMode {
+		// If the controller is in reference mode, ensure that the stored spec is identical to the reference spec at every operation
+		wftSpec, err := woc.fetchWorkflowSpec()
+		if err != nil {
+			return nil, wfv1.Arguments{}, err
+		}
+		if woc.wf.Status.StoredWorkflowSpec.String() != wftSpec.String() {
+			return nil, wfv1.Arguments{}, fmt.Errorf("workflowTemplateRef reference may not change during execution when the controller is in reference mode")
+		}
```

Comment:
Notably, we need to check that the WorkflowTemplate and our own StoredWorkflowSpec have not changed. We do this to prevent attackers from specifying arbitrary code in `wf.Status.StoredWorkflowSpec` and simply running it by using `workflowTemplateRef`.

Reply:
if the can change the status, surely they could have changed the spec?